akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -162,7 +162,8 @@ private[persistence] trait Eventsourced extends Snapshotter with PersistenceStas
   protected def onPersistRejected(cause: Throwable, event: Any, seqNr: Long): Unit = {
     log.warning(
       "Rejected to persist event type [{}] with sequence number [{}] for persistenceId [{}] due to [{}].",
-      event.getClass.getName, seqNr, persistenceId, cause.getMessage)
+      event.getClass.getName, seqNr, persistenceId,
+      s"[${cause.getMessage}]\n${Logging.stackTraceFor(cause)}")
```

Comment:
I see you noticed `log.warning` does not have a `cause` parameter so you included the stacktrace in the message string.

`log.error` actually has a field for the cause. This might play a little nicer with some logging backends, and I think if persisting fails there must be something pretty serious going on, so an `error` might indeed be acceptable here.